soc/arm/st_stm32/stm32l1/soc.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2019 Linaro Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file SoC configuration macros for the STM32L1 family processors.
+ *
+ * Based on reference manual:
+ *   STM32L1X advanced ARM ® -based 32-bit MCUs
+ *
+ * Chapter 2.2: Memory organization
+ */
+
+
+#ifndef _STM32L1_SOC_H_
+#define _STM32L1_SOC_H_
+
+#ifndef _ASMLANGUAGE
+
+#include <stm32l1xx.h>
+
+/* ARM CMSIS definitions must be included before kernel_includes.h.
+ * Therefore, it is essential to include kernel_includes.h after including
+ * core SOC-specific headers.
+ */
+#include <kernel_includes.h>
+
+#include <stm32l1xx_ll_system.h>
+
```

Comment:
Miss
```
#ifdef CONFIG_EXTI_STM32
#include <stm32l1xx_ll_exti.h>
#endif
```

Reply:
@Mani-Sadhasivam, By the way, can I ask you to have a cleaner split of your commits between wrt functionalities addition? I know this requires bit more work on your side, but for instance, it's harder to review when I2C changes are split among multiple commits.